modules/datastore/tests/src/Unit/Plugin/QueueWorker/PostImportResourceProcessorTest.php

```diff
@@ -226,19 +236,130 @@ public function testPostImportProcessItemProcessorError() {
     $this->assertEquals($errors[0], 'Test Error');
   }
 
+  /**
+   * Verify Datastore Drop on Post-Import Error (with drop_config enabled)
+   *
+   * @covers ::postImportProcessItem
+   */
+  public function testDatastoreDropOnPostImportError() {
+    $resource = new DataResource('test.csv', 'text/csv');
+    $resource_processor = (new Chain($this))
+      ->add(ResourceProcessorInterface::class, 'process', new \Exception('Test Error'))
+      ->getMock();
+
+    $container_chain = $this->getContainerChain()
+      ->add(ResourceProcessorCollector::class, 'getResourceProcessors', [$resource_processor])
+      ->add(ResourceMapper::class, 'get', $resource);
+    \Drupal::setContainer($container_chain->getMock());
+
+    $dictionaryEnforcer = PostImportResourceProcessor::create(
+      $container_chain->getMock(), [], '', ['cron' => ['lease_time' => 10800]]
+    );
+    $postImportResult = $dictionaryEnforcer->postImportProcessItem($resource);
+
+    $this->assertEquals($resource->getIdentifier(), $postImportResult->getResourceIdentifier());
+    $this->assertEquals($resource->getVersion(), $postImportResult->getResourceVersion());
+    $this->assertEquals('error', $postImportResult->getPostImportStatus());
+    $this->assertEquals('Test Error', $postImportResult->getPostImportMessage());
+  }
+
+  /**
+   *  Verify Logging on Successful Datastore Drop
+   *
+   * @covers ::postImportProcessItem
+   */
+  public function testLoggingOnSuccessfulDatastoreDrop() {
+    $resource = new DataResource('test.csv', 'text/csv');
+    $resource_processor = (new Chain($this))
+      ->add(ResourceProcessorInterface::class, 'process')
+      ->getMock();
+
+    $container_chain = $this->getContainerChain()
+      ->add(ResourceProcessorCollector::class, 'getResourceProcessors', [$resource_processor])
+      ->add(ResourceMapper::class, 'get', $resource);
+    \Drupal::setContainer($container_chain->getMock());
+
+    $dictionaryEnforcer = PostImportResourceProcessor::create(
+      $container_chain->getMock(), [], '', ['cron' => ['lease_time' => 10800]]
+    );
+    $postImportResult = $dictionaryEnforcer->postImportProcessItem($resource);
+
+    $this->assertEquals($resource->getIdentifier(), $postImportResult->getResourceIdentifier());
+    $this->assertEquals($resource->getVersion(), $postImportResult->getResourceVersion());
+  }
+
+  /**
+   * Verify Exception Handling When Datastore Drop Fails
+   *
+   * @covers ::postImportProcessItem
+   */
+  public function testExceptionHandlingWhenDatastoreDropFails() {
```

Comment:
testExceptionHandlingWhenDatastoreDropFails() is identical 1:1 to testDatastoreDropOnPostImportError(), should one be removed in favor of another?

Reply:
Yes I think we can remove that second one.